### PR TITLE
Fixed course being accessed from a different org site

### PR DESCRIPTION
**Pull request by manan-memon**
This PR fixes the issue of a course from different organization being accessed on LMS site of another one. 

JIRA: https://edlyio.atlassian.net/browse/EDLY-7116

### CODE REVIEW

**Review by muhammadali286 — APPROVED** [review]
We need to test it thoroughly for its ripple affects. Please coordinate with QA accordingly.

File: lms/djangoapps/courseware/courses.py [hunk 1]
```diff
@@ -108,6 +108,18 @@ def get_course_by_id(course_key, depth=0):
...
+def check_if_course_org_in_site_org_filters(course_key):
+    """
+    Given a course key, checks if the course org is in site org.
+
+    If the course is not in site orgs, raises a 404.
+    """
+    current_site_orgs = configuration_helpers.get_current_site_orgs()
+    course = get_course_by_id(course_key)
+    if course.org not in current_site_orgs:
+        raise Http404(u"Course not found: {}.".format(six.text_type(course_key)))
```
Comment (Anas-hameed) [inline lms/djangoapps/courseware/courses.py:120]: Can we move this code to edly app.

File: lms/djangoapps/course_wiki/views.py [hunk 2]
```diff
@@ -38,7 +38,9 @@ def course_wiki_redirect(request, course_id, wiki_path=""):
     as it's home page. A course's wiki must be an article on the root (for
     example, "/6.002x") to keep things simple.
     """
-    course = get_course_by_id(CourseKey.from_string(course_id))
+    course_key = CourseKey.from_string(course_id)
+    course = get_course_by_id(course_key)
+    check_if_course_org_in_site_org_filters(course_key)
```
Comment (Anas-hameed) [inline lms/djangoapps/course_wiki/views.py:43]: Why are we course objects when check_if_course_org_in_site_org_filters is also doing this?

**Review by Anas-hameed — CHANGES_REQUESTED** [review]

File: lms/djangoapps/courseware/courses.py (hunk 1 above)
Comment (manan-memon) [reply lms/djangoapps/courseware/courses.py]: As we discussed, I have changed it to use the `is_course_org_same_as_site_org` from edly app

File: lms/djangoapps/course_wiki/views.py (hunk 2 above)
Comment (manan-memon) [reply lms/djangoapps/course_wiki/views.py]: changed the function to use `is_course_org_same_as_site_org` from edly app, now the course object is only fetched in the view.

**Review by Anas-hameed — APPROVED** [review]
LGTM

**Review by taimoor-ahmed-1 — APPROVED** [review]

**Pull request merged**